src/unix/process.c

```diff
@@ -343,6 +346,424 @@ static void uv__process_child_init(const uv_process_options_t* options,
+
+    if (pipes[fd][1] != -1) {
+      /* If not ignored, make sure the fd is marked as non-blocking */
+      uv__nonblock_fcntl(pipes[fd][1], 0);
+    } else {
+      /* If ignored, redirect to (or from) /dev/null, */
+      err = posix_spawn_file_actions_addopen(
+        actions, 
+        fd, 
+        "/dev/null", 
+        oflags, 
+        mode);
+      if (err != 0)
+        goto error;
+    }
+    
+  }  
+
+  return 0;
+
+error:
+  (void) posix_spawn_file_actions_destroy(actions);
+  return err;
+}
+
+char* uv__spawn_find_path_in_env(char** env) {
+  char** env_iterator = env;
+
+  /* Look for an environment variable called PATH in the 
+   * provided env array, and return its value if found */
+  while (*env_iterator != NULL) {
+    const char* path_name = strstr(*env_iterator, "PATH=");
+    if(path_name != NULL && path_name == *env_iterator) {
+      /* Found "PATH=" at the beginning of the string */
+      return strchr(path_name, '=') + 1;
+    }
+
+    ++env_iterator;
+  }
+
+  return NULL;
+}
+
+int uv__spawn_resolve_and_spawn(const uv_process_options_t* options, 
+                                posix_spawnattr_t* attrs,
+                                posix_spawn_file_actions_t* actions,
+                                pid_t* pid) {
+  const char *p, *z, *path = NULL;
+  size_t l, k;
+  int err = -1;
+
+  /* Short circuit for erroneous case */
+  if (options->file == NULL) 
+    return UV_ENOENT;
+
+  /* The environment for the child process is that of the parent unless overriden 
+   * by options->env */
+  char** env = environ;
+  if (options->env != NULL)
+    env = options->env;
+
+  /* If options->file contains a slash, posix_spawn/posix_spawnp behave
+   * the same, and don't involve PATH resolution at all. Otherwise, if
+   * options->file does not include a slash, but no custom environment is 
+   * to be used, the environment used for path  resolution as well for the 
+   * child process is that of the parent process, so posix_spawnp is the
+   * way to go. */
+  if (strchr(options->file, '/') != NULL || options->env == NULL)
+    return posix_spawnp(pid, options->file, actions, attrs, options->args, env);
+
+  /* Look for the definition of PATH in the provided env */
+  path = uv__spawn_find_path_in_env(options->env);
+
+  /* The following resolution logic (execvpe emulation) is taken from 
+   * https://github.com/JuliaLang/libuv/commit/9af3af617138d6a6de7d72819ed362996ff255d9
+   * and adapted to work around our own situations */
+
+  /* If no path was provided in options->env, use the default value 
+   * to look for the executable */
+  if (!path) 
+    path = _PATH_DEFPATH;
+
+  k = strnlen(options->file, NAME_MAX+1);
+  if (k > NAME_MAX) {
+    errno = ENAMETOOLONG;
+    return -1;
+  }
+
+  l = strnlen(path, PATH_MAX-1)+1;
+
+  for(p = path;; p = z) {
+    /* Compose the new process file from the entry in the PATH
+     * environment variable and the actual file name */
+    char b[l+k+1];
```

Comment:
VLAs are a C99 feature, and not always well supported by compilers today. I think this should be PATH_MAX or malloc?